From d45bae73a2d7b2c1ca9531e898cc30e079c676c9 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <grinberg@compulab.co.il>
Date: Wed, 6 Mar 2013 16:55:36 +0200
Subject: net: FEC: check for valid MAC address

There might be already a valid MAC address provided be the board.
Check if there is and do not override it.

Signed-off-by: Igor Grinberg <grinberg@compulab.co.il>
---
 arch/arm/mach-mx6/mx6_fec.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-mx6/mx6_fec.c b/arch/arm/mach-mx6/mx6_fec.c
index 51be395..690d9f3 100644
--- a/arch/arm/mach-mx6/mx6_fec.c
+++ b/arch/arm/mach-mx6/mx6_fec.c
@@ -45,7 +45,8 @@ static int fec_get_mac_addr(unsigned char *mac)
 
 void __init imx6_init_fec(struct fec_platform_data fec_data)
 {
-	fec_get_mac_addr(fec_data.mac);
+	if (!is_valid_ether_addr(fec_data.mac))
+		fec_get_mac_addr(fec_data.mac);
 
 	if (cpu_is_mx6sl())
 		imx6sl_add_fec(&fec_data);
-- 
1.7.9.5

